lib/client.js

```diff
@@ -137,6 +137,11 @@ Request.prototype.send = function() {
       }
     });
   } else {
+    // label limited to 63 chars and 253 chars max in total length
+    if(this.question.name.length > 253 || !this.question.name.match(/^(?:(?:-|\w|\d\.){1,63}\.)+(?:-|\w|\d\.){1,63}$/)){
+      this.emit('error',new Error(consts.BADNAME));
```

Comment:
I've updated the REGEXP and added more comments to explain what is being matched.